lib/wasix/src/runtime/resolver/outputs.rs

```diff
@@ -27,20 +27,55 @@ pub struct ItemLocation {
     pub package: PackageId,
 }
 
-/// An identifier for a package within a dependency graph.
 #[derive(Debug, Clone, PartialEq, Eq, PartialOrd, Ord, Hash)]
-pub struct PackageId {
-    pub package_name: String,
+pub struct PackageIdent {
+    pub name: String,
     pub version: Version,
 }
 
-impl Display for PackageId {
+impl Display for PackageIdent {
     fn fmt(&self, f: &mut Formatter<'_>) -> fmt::Result {
-        let PackageId {
-            package_name,
+        write!(f, "{}@{}", self.name, self.version)
+    }
+}
+
+#[derive(Debug, Clone, PartialEq, Eq, PartialOrd, Ord, Hash)]
+pub enum PackageId {
+    Named(PackageIdent),
+    HashSha256(String),
```

Comment:
nit: I think that it'd be good if we stuck to one name for the same thing; if we agree, we should either change this to `Sha256Hash` or change the name in `edge-schema` to match this one.